src/features/widgets/store/selectors/spotPrice/useSpotPriceSelectors.tsx

```diff
@@ -0,0 +1,107 @@
+import { useMemo } from 'react';
+import { useStoreContext } from '../../../../../store/useStore';
+import { HistoricalPeriod } from '../../../components/TokenBalanceWidget/TokenBalanceWidgetSettings/TokenBalanceWidgetSettings';
+
+export const nativeToken = 'XTZ';
+export const defaultCurrency = 'USD';
+
+/**
+ * Get native token spot price
+ * @param {string} currency - currency symbols ei USD, EUR
+ * @returns Spot price
+ */
+export const useSelectNativeTokenSpotPrice = (
+  currency?: string
+): string | undefined => {
+  const [state] = useStoreContext();
+  const currencyToken: string = currency ?? defaultCurrency;
```

Comment:
we should not need a 'default' in such a sensitive matter, the 'default currency' should be part of the store. 

I'd suggest we have a `displayFiatCurrency` or so in the state, we will have to implement a selector that updates the display fiat currency -> this will have to refetch all the relevant spot prices upon changing etc.